cdap-client/src/main/java/co/cask/cdap/client/ProgramClient.java

```diff
@@ -75,10 +76,12 @@ public ProgramClient(ClientConfig config, RESTClient restClient, ApplicationClie
     this.applicationClient = applicationClient;
   }
 
+  public ProgramClient(ClientConfig config, RESTClient restClient) {
+    this(config, restClient, new ApplicationClient(config, restClient));
+  }
+
   public ProgramClient(ClientConfig config) {
```

Comment:
Hmm, I swapped the order of the bottom two, but I left the top one on the top, to remain consistent with the other clients' inject method being the topmost.